web/server/network_check.go

```diff
@@ -0,0 +1,325 @@
+package server
+
+import (
+	"context"
+	"encoding/hex"
+	"net"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/pion/stun"
+	"go.viam.com/utils/rpc"
+
+	"go.viam.com/rdk/logging"
+)
+
+// Characterizes the network through a series of general, UDP-based, and TCP-based network
+// checks. Can and should be run asynchronously with server startup to avoid blocking.
+func runNetworkChecks(ctx context.Context) {
+	logger := logging.NewLogger("network-checks")
+	if testing.Testing() {
+		logger.Debug("Skipping network checks in a testing environment")
+		return
+	}
+
+	online := testGeneral(ctx)
+	if !online {
+		logger.Warn("Machine appears to be offline (cannot make connection to app.viam.com); skipping further network checks")
+	}
+
+	if err := testUDP(ctx, logger.Sublogger("udp")); err != nil {
+		logger.Errorw("Error running general network tests", "error", err)
+	}
+
+	if err := testTCP(ctx, logger.Sublogger("tcp")); err != nil {
+		logger.Errorw("Error running general network tests", "error", err)
+	}
+}
+
+// Tests general network connectivity (to app.viam.com.) Stolen from datamanager's
+// sync/connectivity.go code. Returns false is offline and true if online.
+func testGeneral(ctx context.Context) bool {
+	timeout := 5 * time.Second
+	attempts := 1
+	if runtime.GOOS == "windows" {
+		// TODO(RSDK-8344): this is temporary as we 1) debug connectivity issues on windows,
+		// and 2) migrate to using the native checks on the underlying connection.
+		timeout = 15 * time.Second
+		attempts = 2
+	}
+
+	for i := range attempts {
+		// Use DialDirectGRPC to make a connection to app.viam.com instead of a
+		// basic net.Dial in order to ensure that the connection can be made
+		// behind wifi or the BLE-SOCKS bridge (DialDirectGRPC can dial through
+		// the BLE-SOCKS bridge.)
+		ctx, cancel := context.WithTimeout(ctx, timeout)
+		conn, err := rpc.DialDirectGRPC(ctx, "app.viam.com:443", nil)
+		cancel()
+		if err == nil {
+			conn.Close() //nolint:gosec,errcheck
```

Comment:
At this point I `nolint` every `Close` call. The `utils.UncheckError` function is not even helpful IMO because we just don't care about errors.